src/Features/Core/Portable/PullMemberUp/CodeActionAndSolutionGenerator.cs

```diff
@@ -0,0 +1,409 @@
+            AnalysisResult result,
+            Document contextDocument,
+            string title)
+        {
+            if (result.Target.TypeKind == TypeKind.Interface)
+            {
+                var codeGenerationService = contextDocument.Project.LanguageServices.GetRequiredService<ICodeGenerationService>();
+                return CreateAddMembersToInterfaceAction(result, contextDocument, codeGenerationService, title);
+            }
+            else if (result.Target.TypeKind == TypeKind.Class)
+            {
+                return new SolutionChangeAction(
+                    title,
+                    async cancellationToken =>
+                    {
+                        var changedSolution = await GetSolutionAsync(result, contextDocument, cancellationToken).ConfigureAwait(false);
+                        return changedSolution;
+                    });
+            }
+            else
+            {
+                throw ExceptionUtilities.UnexpectedValue(result.Target);
+            }
+        }
+
+        private CodeAction CreateAddMembersToInterfaceAction(
+            AnalysisResult result,
+            Document contextDocument,
+            ICodeGenerationService codeGenerationService,
+            string title)
+        {
+            var options = new CodeGenerationOptions(generateMethodBodies: false, generateMembers: false);
+            var symbolsToPullUp = result.MembersAnalysisResults.
+                Select(analysisResult =>
+                {
+                    if (analysisResult.Member is IPropertySymbol propertySymbol)
+                    {
+                        return CodeGenerationSymbolFactory.CreatePropertySymbol(
+                                propertySymbol,
+                                getMethod: FilterGetterOrSetter(propertySymbol.GetMethod),
+                                setMethod: FilterGetterOrSetter(propertySymbol.SetMethod));
+                    }
+                    else
+                    {
+                        return analysisResult.Member;
+                    }
+                });
+
+            return new DocumentChangeAction(
+                title,
+                async cancellationToken => await codeGenerationService.AddMembersAsync(
+                    contextDocument.Project.Solution, result.Target, symbolsToPullUp, options: options, cancellationToken: cancellationToken));
+        }
+
+        private IMethodSymbol FilterGetterOrSetter(IMethodSymbol getterOrSetter)
+        {
+            return getterOrSetter?.DeclaredAccessibility == Accessibility.Public ? getterOrSetter : null;
+        }
+
+        private SyntaxNode GetAbstractTargetNode(SyntaxNode node, ISymbol symbol, Solution solution)
+        {
+            var generator = SyntaxGenerator.GetGenerator(solution.GetDocument(node.SyntaxTree));
+            var modifiers = DeclarationModifiers.From(symbol).WithIsAbstract(true);
+            return generator.WithModifiers(node, modifiers);
```

Comment:
you don't need the symbol here.  you can get the modifiers from teh node using generator.GetModifiers.